src/main/java/IP/duke/main/Duke.java

```diff
@@ -0,0 +1,61 @@
+package IP.duke.main;
+
+import IP.duke.command.Command;
+
+/**
+ * Represents the duke chat bot, which has the ability to log and track and delete tasks.
+ * 
+ * @author Gordon Yit
+ * @version CS2103T, Semester 2
+ */
+public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    
+    /**
+     * Class constructor.
+     * 
+     * @param fileName name of the storage file.
+     * @param filePath directory of the storage file.
+     */
+    public Duke(String filePath, String fileName) {
+        ui = new Ui();
+        storage = new Storage(filePath, fileName);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+    
+    /**
+     * Executes the duke chat bot.
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while(!isExit) {
```

Comment:
Should there be a space between `while` and `(`?
```suggestion
        while (!isExit) {
```